src/Auth0.php

```diff
@@ -315,22 +315,28 @@ public function __construct(array $config)
             $this->idTokenAlg = $config['id_token_alg'];
         }
 
+        // Make sure the Client ID is included as a valid audience
+        $this->idTokenAud = [ $this->clientId ];
+
         // If a token audience is passed, make sure it's an array.
         if (! empty($config['id_token_aud'])) {
             if (! is_array( $config['id_token_aud'] )) {
                 throw new CoreException('Invalid id_token_aud; must be an array of string values');
             }
 
-            $this->idTokenAud = $config['id_token_aud'];
+            $this->idTokenAud = array_filter( array_merge( $this->idTokenAud, $config['id_token_aud'] ) );
         }
 
+        // Make sure the Domain is included as a valid issuer
+        $this->idTokenIss = [ 'https://'.$this->domain.'/' ];
```

Comment:
when the user uses a "custom domain", will it end up living in the `domain` variable?
Asking because I don't recall now which issuer auth0 uses in that case.